target/linux/ramips/dts/mt7628an_tplink_re200-v2.dts

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,re200-v2", "mediatek,mt7628an-soc";
+	model = "TP-Link RE200 v2";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wps {
+			label = "re200-v2:green:wps";
+			gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi {
+			label = "re200-v2:green:wifi";
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "re200-v2:green:lan";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power: power {
+			label = "re200-v2:green:power";
+			gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi2g_green {
+			label = "re200-v2:green:wifi2g";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		wifi2g_red {
+			label = "re200-v2:red:wifi2g";
+			gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi5g_green {
+			label = "re200-v2:green:wifi5g";
+			gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		wifi5g_red {
+			label = "re200-v2:red:wifi5g";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+
+&state_default {
+	gpio {
+		ralink,group = "p4led_an", "p3led_an", "p2led_an", "p1led_an", "p0led_an", "wled_an", "i2c";
```

Comment:
Buttons will need "uart0" added.

Reply:
Good catch - probably be the reason why they only work every now and then!